services/reports/report.go

```diff
@@ -11,13 +12,19 @@ import (
 	"github.com/mailbadger/app/storage"
 )
 
+const (
+	reportTypeExport = "export"
+)
+
+var (
+	ErrAnotherReportRunning = errors.New("another report running")
+	ErrLimitReached         = errors.New("you reached the limit")
+)
+
 // ReportService represents all report functionalities
 type ReportService interface {
-	IsAnotherReportRunning(*gin.Context, int64) bool
-	IsLimitExceeded(*gin.Context, int64) bool
 	GenerateExportReport(*gin.Context, *entities.Report)
-	GenerateCSV(*gin.Context)
-	GenerateFilename(*gin.Context, int64, string, time.Time) string
+	CreateExportReport(*gin.Context, int64, string, string) (*entities.Report, error)
```

Comment:
```suggestion
	StartReport(*gin.Context, int64, string, string) (*entities.Report, error)
```

Reply:
@FilipNikolovski this method only creates the pending report. GenerateExportReport actually starts generating the report.